### PR TITLE
core.lifetime.emplace(): Properly forward r/lvalue-ness of arguments

### PR DESCRIPTION
**Pull request by kinke**
Some `auto ref` args weren't forwarded.

### CODE REVIEW

File: src/core/lifetime.d [hunk 1]
```diff
@@ -931,51 +945,6 @@ version (CoreUnittest)
     }
 }
 
-/+ these tests can't be performed in druntime, but a mirror still exists in phobos...
```
Comment (kinke) [inline src/core/lifetime.d:934]: These tests are still in `std.conv`. I don't think this comment and the 2 disabled tests provide any value whatsoever here in druntime and see it rather as confusing clutter.

**Review by RazvanN7 — APPROVED** [review]

**Comment by thewilsonator** [comment]
cc @TurkeyMan does this fix https://issues.dlang.org/show_bug.cgi?id=19904 ?

**Comment by TurkeyMan** [comment]
Hard to know. It looks like it's the right path to fixing it... whether every missing call to `forward` has been caught though, I'd need to audit. I have no working project right now to test on.

**Comment by kinke** [comment]
There was this and 2 Phobos PRs: https://github.com/dlang/phobos/pull/7745 and https://github.com/dlang/phobos/pull/7776

**Pull request merged**